### ISSUE TITLE
Train on different Dataset(HVU)

### ISSUE THREAD

**Issue by Joeycho** [facebookresearch/TimeSformer#46]
Hi, thanks for sharing the code.

I am retraining TimeSformer on different Dataset, which is called ["HVU" Dataset](https://holistic-video-understanding.github.io/) for video tagging task.

And I got the following results with the only one data.( the same video and labels for training, validation and test).

So, data(CSV) was given as..

`label1|label2|label3|label4, path_of_video` for training, validation and test.

![validation_error](https://user-images.githubusercontent.com/29337166/122647825-65b11a80-d126-11eb-8fc3-4fba1ee42838.JPG)
![train_error](https://user-images.githubusercontent.com/29337166/122647826-6649b100-d126-11eb-98b7-1c44bc85b141.JPG)

Could I expect overfitting at least in training loss? Or with one data, I cannot expect any relevant results from it?

**Comment by gberta** [comment]
I'm not sure I fully understand your question. However, if you want to train TimeSformer on a multi-label dataset you would need to create cvs lists of the form "video_path label1,label2,label3,label4,label5,...". Afterwards, you would need to write a new dataset class, and add it to lib/dataset folder. 

In this new dataset class, you would read in all of these label values, and then return them in the getitem method (instead of returning a single label). Please take a look at lib/dataset/kinetics.py for a reference on how data loading is done. 

Lastly, you would need to train a model using a sigmoid cross-entropy loss rather than softmax cross-entropy loss.

**Comment by Joeycho** [comment]
@gberta thanks for your answer!

Yes, I already did exactly what you said. The key point was binary cross-entropy loss rather than softmax cross-entropy loss. I refer the setup for Charades dataset from SlowFast repository.

My question was whether the model could be overfit on super small tiny data such as 5 training videos, 2 per validation and 2 per tests.

It seems not working with the tiny dataset. I kept seeing loss graph like the above around 4.29.

However, after some hypertuning and targeting on 1% dataset(4300 training set, 100 validation 100 test) rather than tiny set, 

It worked as I expected.

![Train_value](https://user-images.githubusercontent.com/29337166/122945586-f67f3480-d378-11eb-8f79-9c0988b3d0d4.JPG)
